internal/controller/etcdcluster_controller.go

```diff
@@ -199,6 +207,28 @@ func (r *EtcdClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return r.updateStatus(ctx, instance)
 }
 
+// checkAndDeleteStatefulSetIfNecessary deletes the StatefulSet if the specified storage size has changed.
+func (r *EtcdClusterReconciler) checkAndDeleteStatefulSetIfNecessary(ctx context.Context, state *observables, instance *etcdaenixiov1alpha1.EtcdCluster) error {
+	for _, volumeClaimTemplate := range state.statefulSet.Spec.VolumeClaimTemplates {
+		if volumeClaimTemplate.Name != "data" {
+			continue
+		}
+		currentStorage := volumeClaimTemplate.Spec.Resources.Requests[corev1.ResourceStorage]
+		desiredStorage := instance.Spec.Storage.VolumeClaimTemplate.Spec.Resources.Requests[corev1.ResourceStorage]
+		if desiredStorage.Cmp(currentStorage) != 0 {
```

Comment:
I think we should check in validating webhook that new desired size is greater or equal to previous

Reply:
sure, done